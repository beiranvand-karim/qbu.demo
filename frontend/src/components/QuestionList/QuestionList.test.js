/* global describe, it, expect, jest */
import React from 'react'
import {shallow} from 'enzyme'
import {QuestionList} from './'
import renderer from 'react-test-renderer'
import {MemoryRouter} from "react-router-dom"

describe('<QuestionList />',() => {
   const questionList = [
      {
         "id": 7,
         "title": "this is a title",
         "text": "dkg;dkg;d;g",
         "prize": 45000,
         "userId": 7,
         "username": "admin"
      },
      {
         "id": 8,
         "title": "who is the best soccer player?",
         "text": "we want to know who is curently the best soccer player ther is?",
         "prize": 2000,
         "userId": 7,
         "username": "admin"
      }
   ];

   it('should render without crashing', () => {
      shallow(<QuestionList questionList={questionList} />)
   });

   it('should matches the snapshot', () => {
      const tree = renderer.create(<MemoryRouter><QuestionList questionList={questionList}/></MemoryRouter>).toJSON();
      expect(tree).toMatchSnapshot()
   });
});