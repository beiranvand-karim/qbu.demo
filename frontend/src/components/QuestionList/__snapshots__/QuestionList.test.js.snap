// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuestionList /> should matches the snapshot 1`] = `
Array [
  <div
    className="sc-bdVaJa gDhQkR"
  >
    <div
      className="sc-bwzfXH eiaxpn"
    >
      <div
        className="jumbotron"
      >
        <h1
          className="display-6"
        >
          this is a title
        </h1>
        <p
          className="lead"
        >
          admin
        </p>
        <p
          className="lead"
        >
          45000
        </p>
        <hr
          className="my-2"
        />
        <p>
          dkg;dkg;d;g
        </p>
        <p
          className="lead"
        >
          <button
            aria-label={null}
            className="btn btn-primary"
            onClick={[Function]}
            type="button"
          >
            visit question
          </button>
        </p>
      </div>
    </div>
  </div>,
  <div
    className="sc-bdVaJa gDhQkR"
  >
    <div
      className="sc-bwzfXH eiaxpn"
    >
      <div
        className="jumbotron"
      >
        <h1
          className="display-6"
        >
          who is the best soccer player?
        </h1>
        <p
          className="lead"
        >
          admin
        </p>
        <p
          className="lead"
        >
          2000
        </p>
        <hr
          className="my-2"
        />
        <p>
          we want to know who is curently the best soccer player ther is?
        </p>
        <p
          className="lead"
        >
          <button
            aria-label={null}
            className="btn btn-primary"
            onClick={[Function]}
            type="button"
          >
            visit question
          </button>
        </p>
      </div>
    </div>
  </div>,
]
`;
